### PR TITLE
Work around the fact that crust gives us multiple bootstrap endpoints

### PR DESCRIPTION
**Pull request by inetic**
[Review on Reviewable](https://reviewable.io/reviews/maidsafe/routing/543)

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @benjaminbollen

(maidsafe_highfive has picked a reviewer for you, use r? to override)

**Comment by benjaminbollen** [comment]
restarted all CI builds; as routing master was not compiling - but now is

File: src/messages.rs [hunk 1]
```diff
@@ -302,9 +302,9 @@ impl SignedMessage {
         utils::decode::<RoutingMessage>(&self.encoded_body)
     }
 
-    pub fn encoded_body(&self) -> &Vec<u8> {
```
Comment (benjaminbollen) [inline src/messages.rs]: these functions have been marked with #[allow(dead_code)]; sorry for the conflict

File: src/relay.rs [hunk 2]
```diff
@@ -124,9 +124,9 @@ impl RelayMap {
     }
 
     /// Returns true if we already have a name associated with this endpoint.
-    pub fn contains_endpoint(&self, relay_endpoint: &Endpoint) -> bool {
```
Comment (benjaminbollen) [inline src/relay.rs]: same here, marked with #[allow(dead_code)]

**Comment by benjaminbollen** [comment]
can you rebase this to master? thx!

File: src/routing_node.rs [hunk 3]
```diff
@@ -82,100 +80,67 @@ impl<F, G> RoutingNode<F, G> where F : Interface + 'static,
...
-                                                    }
-                                                },
-                                                _ => continue, // only care about a Node
-                                            }
+                    //println!("Event::NewMessage({:?}, bytes)", endpoint);
+
+                    let self_id = self.id.clone();
+
+                    if self.bootstraps.contains(&endpoint) {
+                        if let Ok(signed_message) = decode::<SignedMessage>(&bytes) {
```
Comment (benjaminbollen) [inline src/routing_node.rs]: nice syntax

File: src/routing_node.rs [hunk 4]
```diff
@@ -82,100 +80,67 @@ impl<F, G> RoutingNode<F, G> where F : Interface + 'static,
...
+                        }
+                        else if let Ok(he_is_msg) = decode::<IAm>(&bytes) {
+                            match he_is_msg.address {
+                                Address::Node(node_name) => {
+                                    info!("Name of our relay node is {:?}", node_name);
+                                    our_bootstrap = Some((endpoint.clone(), node_name.clone()));
                                     if !sent_name_request {
-                                        // now send a PutPublicId request
-                                        let our_public_id = PublicId::new(&self.id);
+                                        sent_name_request = true;
```
Comment (benjaminbollen) [inline src/routing_node.rs]: later on this can be a state-variable per bootstrap connection in the BTreeMap

**Comment by benjaminbollen** [comment]
looks good to me, but we should drop all connections in connection_manager that were established, but not utilised in the end; as we still only pass a single (bootstrap_endpoint, bootstrap_name) to membrane, and membrane will only close that connection (after RoutingTable connections are established)

**Pull request merged**